packages/editor/src/components/provider/index.js

```diff
@@ -194,8 +194,11 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 					postContext.postType = 'post';
 				} else if ( post.slug.split( '-' )[ 0 ] === 'single' ) {
 					// If the slug is single-{postType}, infer the post type from the slug.
-					const postTypesSlugs =
-						postTypes?.map( ( entity ) => entity.slug ) || [];
+					const postTypesSlugs = useMemo( () => {
+						return (
+							postTypes?.map( ( entity ) => entity.slug ) || []
+						);
+					}, [] );
```

Comment:
Using hooks inside condition is against [React Hook rules](https://react.dev/warnings/invalid-hook-call-warning), and I don't think nesting `useMemo` calls is a good idea.

We should move `postTypesSlugs` map outside of current memo function.